src/ocf_lru.c

```diff
@@ -588,6 +592,69 @@ static void ocf_lru_clean_end(void *private_data, int error)
 	env_refcnt_dec(&ctx->counter);
 }
 
+static inline ocf_cache_line_t lru_iter_repart_next(struct ocf_lru_iter *iter,
+		struct ocf_part *dst_part)
+{
+	ocf_cache_t cache = iter->cache;
+	struct ocf_part *part = iter->part;
+	uint32_t curr_lru;
+	ocf_cache_line_t  cline;
+	struct ocf_lru_list *list;
+	ocf_core_id_t core_id;
+	ocf_core_t core;
+
+	if (dst_part == part)
+		return end_marker;
+
+	do {
+		curr_lru = _lru_next_lru(iter);
+
+		ocf_metadata_lru_wr_lock(&cache->metadata.lock, curr_lru);
+
+		list = ocf_lru_get_list(part, curr_lru, iter->clean);
+
+		cline = list->tail;
+		while (cline != end_marker && !ocf_cache_line_try_lock_wr(
+				iter->c, cline)) {
+			cline = ocf_metadata_get_lru(cache, cline)->prev;
+		}
+
+		if (cline != end_marker) {
+			core_id = ocf_metadata_get_core_id(cache, cline);
+			core = ocf_cache_get_core(cache, core_id);
+			if (metadata_test_dirty(cache, cline))
+				ocf_cleaning_purge_cache_block(cache, cline);
+
+			ocf_lru_repart_locked(cache, cline, part, dst_part);
```

Comment:
I don't think `ocf_lru_repart_locked()` is a good fit for repartition during a managment operation because every reparted cache line will be marked as hot and will trigger unnecessary rebalancing on the default ioclass.